lib/schema.ex

```diff
@@ -17,20 +17,24 @@ defmodule Rajska.Schema do
     Field.t(),
     module()
   ) :: [Middleware.spec(), ...]
-  def add_query_authorization(
-    [{{QueryAuthorization, :call}, config} = query_authorization | middleware] = _middleware,
-    %Field{name: query_name},
-    authorization
-  ) do
-    validate_query_auth_config!(config, authorization, query_name)
-
-    [query_authorization | middleware]
-  end
+  def add_query_authorization(middleware, %Field{name: query_name}, authorization) do
+    middleware
+    |> Enum.find(&find_middleware/1)
+    |> case do
+      {{QueryAuthorization, :call}, config} ->
+        validate_query_auth_config!(config, authorization, query_name)
+
+      {{Absinthe.Resolution, :call}, _config} ->
+        raise "No permission specified for query #{query_name}"
+    end
 
-  def add_query_authorization(_middleware, %Field{name: name}, _authorization) do
-    raise "No permission specified for query #{name}"
+    middleware
   end
 
+  def find_middleware({{QueryAuthorization, :call}, _config}), do: true
```

Comment:
from the discussion in https://github.com/jungsoft/rajska/pull/27#issuecomment-678452894